lib/server/services/qualityControlFlag/QcFlagService.js

```diff
@@ -331,19 +331,26 @@ class QcFlagService {
                 const createdFlag = await QcFlagRepository.findOne(this.prepareQueryBuilder().where('id').is(newInstance.id));
 
                 const flagInclude = [];
+                let synchronousQcWhereClause = null;
                 if (dataPass) {
                     await createdFlag.addDataPass(dataPass);
                     flagInclude.push({ association: 'dataPasses', where: { id: dataPass.id }, required: true });
-                }
-                if (simulationPass) {
+                } else if (simulationPass) {
                     await createdFlag.addSimulationPass(simulationPass);
                     flagInclude.push({ association: 'simulationPasses', where: { id: simulationPass.id }, required: true });
+                } else {
+                    flagInclude.push({ association: 'dataPasses', required: false });
+                    flagInclude.push({ association: 'simulationPasses', required: false });
+                    synchronousQcWhereClause = { '$flag->dataPasses.id$': null, '$flag->simulationPasses.id$': null };
                 }
+                // Else synchronous QC flag is created
 
                 const effectivePeriodsToBeUpdated = await QcFlagEffectivePeriodRepository.findAll({
+                    subQuery: false,
                     where: {
                         from: { [Op.lt]: createdFlag.to },
                         to: { [Op.gt]: createdFlag.from },
+                        ...synchronousQcWhereClause ?? {},
```

Comment:
Simple question: isn't there a way to simplify this?
Something like putting the include all the time, and something like
`{ '$flag->dataPasses.id$': dataPass?.id ?? null, '$flag->simulationPasses.id$': simulationPass?.id ?? null };`

Reply:
not really, if you create for data pass, reference to simulation passes column must not appear there, the same when creating for simulation passes, but when creating synchronous flag, both must appear in where clause